### DISCUSSION TITLE
UX Bug: Unable to modify an instance variable when the default is structure

### DISCUSSION THREAD

**Discussion by krunkster** [4ian/GDevelop#2351] [Enhancement ideas (and their technical discussions)]
## Description
If you have an object with an instance variable with a default value that is in a structure, when you try to change that value on an instance of that object in the properties panel the field associated with all those instance variables in the structure are immutable.

## To Reproduce
Add any object to a scene
Right click on the object in the object panel and choose "Edit object variables"
Click the "+ ADD" button to add a new variable named "Variable"
Click the "+" icon next to "Variable" to add a new child variable called "ChildVariable"
Drag the object from the panel to the scene to create a new instance
Open the Properties Panel if not already
Click on the instance of the object in the scene
See that "Variable" and "ChildVariable" are grayed out and immutable
Furthermore: Click on the "pop out" icon and see that "Variable" is entirely missing.

## Workaround
Do not define any default instance variables using a structure
Manually create the structure in the properties editor on a instance of the object
Copy that instance to maintain the "default" values in the structure

## Other details
Windows
5.0 b100

**Comment by 4ian** [comment]
It's indeed true that you there is no way to override the values of a structure children - basically the instance variables will override the entire variable and all the children, without a way to only change children.
It's a missing feature in a way, so I'll flag it as so.

**Comment by AlexandreSi** [comment]
See https://github.com/4ian/GDevelop/issues/2707 for more details
